serverless.yml

```diff
@@ -67,4 +67,57 @@ custom:
     excludeFiles: src/**/*.test.js # Provide a glob for files to ignore
   # https://www.serverless.com/plugins/serverless-offline#usage-with-webpack - See Usage and command line options
   serverless-offline:
-    httpPort: ${env:AWS_SERVER_PORT, 3001}
+    httpPort: ${env:AWS_SERVER_PORT, 3020}
+  dynamodb:
+    start:
+      port: ${env:DYNAMO_PORT, 8020}
+      inMemory: true
```

Comment:
One thing we can do I think is have a dynamic config for launching dynamoDB locally since its only used in our local development.

This will do the trick I think
(following this example: https://www.serverless.com/framework/docs/providers/aws/guide/variables#reference-variables-in-javascript-files)

```diff
diff --git a/serverless.yml b/serverless.yml
index 8c9f137..ff36ade 100644
--- a/serverless.yml
+++ b/serverless.yml
@@ -72,9 +72,9 @@ custom:
     start:
       port: ${env:DYNAMO_PORT, 8020}
       inMemory: true
-      migrate: false
-      seed: false
-      noStart: true
+      migrate: ${file(./config/setConfig.js):slsDBConfig.migrate}
+      seed: ${file(./config/setConfig.js):slsDBConfig.seed}
+      noStart: ${file(./config/setConfig.js):slsDBConfig.noStart}
     seed:
       trailer-registration:
         sources:
@@ -84,9 +84,9 @@ custom:
       - local
       - dev
       - test
-  
+
 resources:
-   Resources:
+  Resources:
     TrailerRegistration:
       Type: AWS::DynamoDB::Table
       Properties:
```

```js
// ./config/setConfig.js

const isLocal = process.env.NODE_ENV === 'local';
// used locally when developping the service so DynamoDB starts
// The CI pipeline has its own instance of dynamo and only consumes the bundled file
module.exports.slsDBConfig = (serverless) => {
  serverless.cli.consoleLog(`Creating config for ${process.env.NODE_ENV}`);
  return {
    migrate: isLocal ? true : false,
    seed: isLocal ? true : false,
    noStart: isLocal ? false : true,
  };
};
```

Reply:
Will this have an impact on the build pipeline?

Reply:
I think it shouldn't as we don't use serverless to deploy and don't believe in the pipeline we have such env variable, to my knowledge (and last time I had a peak :) ), we have BRANCH, BUCKET and one i forgot. We can rerun the job and see if the integration tests fail